ui-core/src/components/inputs/checkbox/CheckboxList.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { CheckboxState, checkboxListItem } from './type';
+import Checkbox from './Checkbox';
+
+import cx from 'classnames';
+
+export type CheckboxListProps = {
+  items: checkboxListItem[];
```

Comment:
use CheckboxTreeItem, the non flatten type
```suggestion
  items: CheckboxTreeItem[];
```